fastapi/applications.py

```diff
@@ -117,6 +119,10 @@ def __init__(
         self.extra = extra
         self.dependency_overrides: Dict[Callable[..., Any], Callable[..., Any]] = {}
 
+        self.liveness_probe_url = liveness_probe_url
+        self.readiness_probe_url = readiness_probe_url
+        self.probes_state: State = State({"alive": True, "ready": True})
```

Comment:
What is the reason behind using the State object from `Starlette` and not a Pydantic `dataclass`, or anything that constrains the types of this object to `alive` and `ready`?

Reply:
I don't have a good answer. I think the moment I choose to do that I was thinking about "more customizable attributes" (some changes are needed for this to be true). 

About the dataclass, I'm not sure because we'll need to install the dataclasses backport package for 3.6. It's not really a problem by itself, is just that it will be the first here. 😅

But anyway, your comment is relevant. Thank you :)

I'll apply modifications on the PR once I have more feedbacks about the whole idea. Because as stated in the description, it's kind of a prototype.

Reply:
@klaa97 I don't see a problem here. If you check on fastapi's code you will not see any dataclasses declared. So I would say LGTM.